source/util/FeatureRegistry.h

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef AWS_IOT_DEVICE_CLIENT_FEATURE_REGISTRY_H
+#define AWS_IOT_DEVICE_CLIENT_FEATURE_REGISTRY_H
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "../Feature.h"
+#include "../config/Config.h"
+
+namespace Aws
+{
+    namespace Iot
+    {
+        namespace DeviceClient
+        {
+            namespace Util
+            {
+
+                /**
+                 * @brief A class to keep track of the features currently running on the device client
+                 */
+                class FeatureRegistry
+                {
+                  public:
+                    FeatureRegistry() = default;
+                    std::shared_ptr<Feature> get(const std::string &name) const;
+                    void add(std::shared_ptr<Feature> feature);
+                    void remove(const std::string &name);
+                    int getSize();
```

Comment:
`getSize()`, `stopAll`, and `startAll` can all be `const` member functions eg add `const` suffix.